lib/http_server/router.ex

```diff
@@ -18,23 +19,24 @@ defmodule HttpServer.Router do
     |> Response.format()
   end
 
-  def find_handler(routes \\ Routes, request) do
-    case request.method do
-      "HEAD" ->
-        %{request | method: "GET"}
-        |> routes.handle()
-      _ ->
-        routes.handle(request)
-    end
+  def find_handler(routes \\ Routes, request)
+
+  def find_handler(routes, %Request{method: "HEAD"} = request) do
+    %Request{request | method: "GET"}
+    |> routes.handle()
```

Comment:
This is looking good.  Since we know this is for `HEAD`, why not just strip the headers right here and simplify things.
```suggestion
  def find_handler(routes \\ Routes, request)

  def find_handler(routes, %Request{method: "HEAD"} = request) do
    response = %Request{request | method: "GET"} |> routes.handle()
    
    %Response{ response | body: nil }
```

Reply:
I see you have this in the `Response`.  I think that actually makes more sense, it should not be the routers job to strip the body.